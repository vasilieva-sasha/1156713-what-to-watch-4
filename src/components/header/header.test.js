import React from "react";
import renderer from "react-test-renderer";
import Header from "./header";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from './../../reducer/name-space';
import {Router} from 'react-router-dom';
import history from './../../history';

const mock = {
  avatar: ``
};

const mockStore = configureStore([]);
const store = mockStore({
  [NameSpace.USER]: {
    authorizationStatus: `NO_AUTH`,
    authInfo: null,
  },
  [NameSpace.APP]: {
    currentPage: `MAIN`
  }
});

describe(`HeaderComponent`, () => {
  it(`Header renders correctly with no-auth`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <Provider store={store}>
            <Header authorizationStatus={store.authorizationStatus} authInfo={mock} currentPage={store.currentPage}/>
          </Provider>
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Header renders correctly with no-auth`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <Provider store={store}>
            <Header authorizationStatus={`AUTH`} authInfo={store.authInfo} currentPage={store.currentPage}/>
          </Provider>
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
