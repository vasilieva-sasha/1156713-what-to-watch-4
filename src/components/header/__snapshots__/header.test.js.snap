// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Header renders correctly with no-auth 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="#"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </div>
</header>
`;

exports[`HeaderComponent Header renders correctly with no-auth 2`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="#"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </div>
</header>
`;
